doc/OnlineDocs/contributed_packages/community.rst

```diff
@@ -0,0 +1,274 @@
+    communities within a model. Thus, it can be useful to call `detect_communities(model, with_objective=False)`
+    on such a model to see whether isolating the other components of the model provides any new insights.
+
+External Packages
+-----------------
+The community detection package relies on two external packages, the NetworkX package and the Louvain community
+detection package. Both of these packages can be installed at the following URLs (respectively):
+
+https://pypi.org/project/networkx/
+
+https://pypi.org/project/python-louvain/
+
+The pip install and conda install commands are included below as well:
+
+
+>>> pip install networkx
+>>> pip install python-louvain
+>>> conda install -c anaconda networkx
+>>> conda install -c conda-forge python-louvain
+
+Usage Examples
+--------------
+
+Let's take the following model as an example (taken from `Allman et al, 2019`_) (the necessary
+imports to run all of the code examples are included below):
+
+.. _Allman et al, 2019: https://doi.org/10.1007/s11081-019-09450-5
+
+    >>> from pyomo.contrib.community_detection.detection import detect_communities, visualize_model_graph, \
+    >>>     stringify_community_map, generate_model_graph
+    >>> from pyomo.contrib.mindtpy.tests.eight_process_problem import EightProcessFlowsheet
+    >>> from pyomo.core import ConcreteModel, Var, Constraint
+    >>> import matplotlib.pyplot as plt
+    >>> import networkx as nx
+    >>>
+    >>> def decode_model_1():
+    >>>     model = m = ConcreteModel()
+    >>>     m.x1 = Var(initialize=-3)
+    >>>     m.x2 = Var(initialize=-1)
+    >>>     m.x3 = Var(initialize=-3)
+    >>>     m.x4 = Var(initialize=-1)
+    >>>     m.c1 = Constraint(expr=m.x1 + m.x2 <= 0)
+    >>>     m.c2 = Constraint(expr=m.x1 - 3 * m.x2 <= 0)
+    >>>     m.c3 = Constraint(expr=m.x2 + m.x3 + 4 * m.x4 ** 2 == 0)
+    >>>     m.c4 = Constraint(expr=m.x3 + m.x4 <= 0)
+    >>>     m.c5 = Constraint(expr=m.x3 ** 2 + m.x4 ** 2 - 10 == 0)
+    >>>     return model
+    >>> model = m = decode_model_1()
+    >>>
+    >>> print(detect_communities(model, type_of_community_map='b'))
+
+Here is the output of the `detect_communities` call above:
+
+    >>> # Output:
+    >>> {0: ([<pyomo.core.base.constraint.SimpleConstraint object at 0x0000028DA74BB588>,
+    >>>       <pyomo.core.base.constraint.SimpleConstraint object at 0x0000028DA74BB5F8>],
+    >>>      [<pyomo.core.base.var.SimpleVar object at 0x0000028DA74BB3C8>,
+    >>>       <pyomo.core.base.var.SimpleVar object at 0x0000028DA74BB438>]),
+    >>>  1: ([<pyomo.core.base.constraint.SimpleConstraint object at 0x0000028DA74BB668>,
+    >>>       <pyomo.core.base.constraint.SimpleConstraint object at 0x0000028DA74BB6D8>,
+    >>>       <pyomo.core.base.constraint.SimpleConstraint object at 0x0000028DA74BB748>],
+    >>>      [<pyomo.core.base.var.SimpleVar object at 0x0000028DA74BB4A8>,
+    >>>       <pyomo.core.base.var.SimpleVar object at 0x0000028DA74BB518>])}
```

Comment:
Don't use `>>>` for output. See https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html for how to setup and run a doctest.